csv-validator-ui/src/main/scala/uk/gov/nationalarchives/csv/validator/ui/ScalaSwingHelpers.scala

```diff
@@ -17,12 +17,24 @@ import java.nio.file.Path
 import scala.swing.Dialog.Message
 import java.io.IOException
 import javax.swing.JTextField
+import scala.util.{Failure, Success, Try}
 
 /**
  * Some simple helpers to ease
  * the use of scala.swing
  */
 object ScalaSwingHelpers {
+  private def handleSelectedFile(dialogAction: Result.Value, fileChooser: FileChooser, result: Path => Try[String], dialogText: String): Unit =
+    dialogAction match {
+      case Result.Approve =>
+        result(fileChooser.selectedFile.toPath) match {
+          case Failure(e) =>
+            e.printStackTrace()
+            Dialog.showMessage(fileChooser, s"${e.getClass.getName}: ${e.getMessage}", s"Unable to ${dialogText.toLowerCase()} file", Message.Error)
+          case Success(_) =>
+        }
+      case Result.Cancel =>
```

Comment:
should these ignored cases be just `case _`  
e.g. For the outer match, `Result.Error` is never matched. Not sure if that causes any issue further down.

Reply:
Yes, I think without `()`, they look weird; I'll add it.

Reply:
Sorry, I was referring to 2 things, 
1) We do not have catch all 
2) Is there any value in using a typed matching?

`case Success(_) =>` can change to `case _ => ()` and 
`case Result.Cancel =>` can change to `case _ => ()` 

Basically, other than the specific handled case, anything else is just `()`